### PR TITLE
Support highlighting API

### PR DESCRIPTION
**Pull request by mnunberg**
Might still need some tests

### CODE REVIEW

**Comment by mnunberg** [comment]
actually.. I'm writing tests and there are indeed a few bugs :)

File: redisearch/query.py [hunk 1]
```diff
@@ -51,6 +51,48 @@ def return_fields(self, *fields):
         self._return_fields = fields
         return self
 
+    def summarize(self, *fields, **options):
```
Comment (dvirsky) [inline redisearch/query.py]: I prefer that the kwargs will have the defaults already. I hate it when python code sends you on an RTFM mission to decipher what **kwargs actually mean...

File: redisearch/query.py [hunk 2]
```diff
@@ -51,6 +51,48 @@ def return_fields(self, *fields):
...
+            args += ['LEN', str(options['context_len'])]
+        if 'num_frags' in options:
+            args += ['FRAGS', str(options['num_frags'])]
+        if 'sep' in options:
+            args += ['SEPARATOR', options['sep']]
+
+        self._summarize_fields = args
+        return self
+
+    def highlight(self, *fields, **options):
```
Comment (dvirsky) [inline redisearch/query.py]: Again, provide keyword arguments with sane defaults here...

**Comment by dvirsky** [comment]
Looks good besides the **options thing which I would rewrite as defaults. 

Please also regenerate the docs with `gendocs.py redisearch` and pipe the resulting MD to the redisearch docs python api markdown.

Thanks!

File: redisearch/query.py (hunk 1 above)
Comment (mnunberg) [reply redisearch/query.py]: The problem with this is that it disallows me from specifying multiple fields, and the user must then do one field at a time. I guess it's acceptable, but annoying.

**Comment by mnunberg** [comment]
Using `highlight-api` branch instead. Will reopen PR

**Pull request closed without merging**